client/src/Components/Modal/index.tsx

```diff
@@ -0,0 +1,76 @@
+/* eslint-disable jsx-a11y/no-noninteractive-element-interactions*/
+/* eslint-disable jsx-a11y/click-events-have-key-events*/
+
+import FocusTrap from 'focus-trap-react'
+import React, { useCallback, useEffect } from 'react'
+import ReactDOM from 'react-dom'
+
+export interface ModalContainerProps {
+  isOpen: boolean
+  header: string
+  children: string
+  onClose: () => void
+}
+
+const ModalContainer: React.FC<ModalContainerProps> = ({
+  isOpen,
+  header,
+  children,
+  onClose
+}) => {
+  const escFunction = useCallback(
+    e => {
+      if (e.keyCode === 27) {
+        onClose()
+      }
+    },
+    [onClose]
+  )
+  useEffect(() => {
+    document.addEventListener('keydown', escFunction)
+    return () => {
+      document.removeEventListener('keydown', escFunction)
+    }
+  }, [escFunction])
+  if (isOpen) {
+    const portalDiv = document.getElementById('portal')
+    return portalDiv
+      ? ReactDOM.createPortal(
```

Comment:
Instead of using a ternary and an if statement, it would be easier to assign the element to a variable and the `#portal` element to a ref and then return the element when required. 
```
const ref = useRef()

useEffect(() => {
    ref.current = document.getElementById('portal')
}, [])

const modal = (
    <FocusTrap>
...
    </FocusTrap>
)
return isOpen ? ReactDOM.createPortal(modal, ref.current) : null
```